### PR TITLE
Add dataflow traces in PRs and MRs documentation, remove figure numbers

### PR DESCRIPTION
**Pull request by adamkvitek**
When building locally with `yarn start`, go to:
- http://localhost:3000/docs/semgrep-code/semgrep-pro-engine-data-flow/
- http://localhost:3000/docs/semgrep-cloud-platform/github-pr-comments/#dataflow-traces-in-pr-comments

This documentation is rendered at the following places:
Screenshot 2023-05-24 at 17 12 47
Screenshot 2023-05-24 at 17 14 41

# Thanks for improving Semgrep Docs 😀

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR
- [ ] This change has no security implications or else you have pinged the security team
- [ ] Redirects are added if the PR changes page URLs
- [ ] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### CODE REVIEW

File: docs/semgrep-cloud-platform/github-pr-comments.md [hunk 1]
```diff
@@ -56,4 +57,21 @@ Only rules in the **Comment** and **Block** columns of your [Rule board](https:/
...
+**Figure** An inline GitHub pull request comment with dataflow traces.
+
+<DisplayTaintedDataIntro />
+
+### Viewing the path of tainted data in PR comments
+
+To enable dataflow traces feature in your CI pipeline, fulfill the following prerequisites:
+
+:::info Prerequisites
+- Enable GitHub pull request comments from Semgrep.
```
Comment (ajbt200128) [inline docs/semgrep-cloud-platform/github-pr-comments.md:72]: Dataflow trace graphs will also work in gitlab PR comments, just not bitbucket

Comment (adamkvitek) [reply docs/semgrep-cloud-platform/github-pr-comments.md:72]: Ok! I will update the PR accordingly.

File: src/components/procedure/_semgrep-code-display-tainted-data.mdx [hunk 2]
```diff
@@ -2,7 +2,7 @@
 Not all Semgrep rules or rulesets make use of taint tracking. Ensure that you have a ruleset, such as the **default ruleset** added in your **[Rule Board](https://semgrep.dev/orgs/-/board)**. If this ruleset is not added, go to [https://semgrep.dev/p/default](https://semgrep.dev/p/default), and then click **Add to Rule Board**. You can add rules that use taint tracking from [Semgrep Registry](https://semgrep.dev/explore).
 :::
 
-To view details of tainted data, follow these steps:
+To view detailed path of tainted data with dataflow traces, follow these steps:
```
Comment (s-santillan) [inline src/components/procedure/_semgrep-code-display-tainted-data.mdx:5]:
```suggestion
To view a detailed path of tainted data with dataflow traces, follow these steps:
```

**Review by s-santillan — APPROVED** [review]
lgtm!

File: docs/semgrep-cloud-platform/github-pr-comments.md (hunk 1 above)
Comment (adamkvitek) [reply docs/semgrep-cloud-platform/github-pr-comments.md]: A section about GitLab MR comments added. We might need to add a screenshot of dataflow traces in GitLab.

**Review by ajbt200128 — APPROVED** [review]

**Pull request merged**